src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs

```diff
@@ -318,22 +322,31 @@ private void OutputToOutputWindow(string message)
                 }
             }
 
-            foreach (var reference in commandLineArguments.MetadataReferences)
+            foreach (var reference in commandLineArguments.ResolveMetadataReferences(project.CurrentCompilationOptions.MetadataReferenceResolver))
             {
+                // References that don't exist on disk will come back prepended with "Unresolved: ".  We'll strip that off 
+                // and still add them, so that if they appear, we'll load them (maybe they are outputs of other parts of the build).
+                const string unresolved = "Unresolved: ";
```

Comment:
Yeah, probably safer to trycast to UnresolvedMetadataReference and access Reference property. If trycast fails, hard cast to PortableExecutableReference and access FilePath?